### PR TITLE
feat: Add a setting to respect system alerts while detecting active apps

### PR DESCRIPTION
**Pull request by mykola-mokhnach**
Currently we detect the app under test as active if XCTest returns RunningInForeground state for it. In case the app is covered by a system alert from springboard this might be confusing as we cannot interact with it unless an alert is properly handled. The new setting called `respectSystemAlerts` being set to `true` (by default it is `false`) forces WDA to verify the presence of alerts shown by springboard and return the latter if an alert is shown. This affects the performance of active app detection, but might be more convenient for writing test script (e.g. eliminates the need of proactive switching between system and custom apps).

### CODE REVIEW

File: WebDriverAgentLib/Utilities/FBSettings.m [hunk 1]
```diff
@@ -34,3 +34,4 @@
 NSString* const FB_SETTING_WAIT_FOR_IDLE_TIMEOUT = @"waitForIdleTimeout";
 NSString* const FB_SETTING_ANIMATION_COOL_OFF_TIMEOUT = @"animationCoolOffTimeout";
 NSString* const FB_SETTING_MAX_TYPING_FREQUENCY = @"maxTypingFrequency";
+NSString* const FB_SETTING_RESPECT_SYSTEM_ALERTS = @"respectSystemAlerts";
```
Comment (mykola-mokhnach) [inline WebDriverAgentLib/Utilities/FBSettings.m:37]: I was not quite sure about the setting name. Maybe you'd have better proposals

**Review by KazuCocoa — COMMENTED** [review]
the change itself lg, let me think of the naming

**Review by KazuCocoa — APPROVED** [review]
lgtm. "system alert" naming is reasonable to imagine this setting.

**Review by eglitise — APPROVED** [review]
Thanks for doing this. From what I can tell this only handles `XCUIElementTypeAlert` elements, which should cover most cases, but I vaguely recall (from my own projects) that there may be some other element types as well (e.g. picker wheels or file sharing modals). I will check this on Monday/Tuesday, but those can also be handled in a separate PR if necessary.

**Comment by mykola-mokhnach** [comment]
> Thanks for doing this. From what I can tell this only handles `XCUIElementTypeAlert` elements, which should cover most cases, but I vaguely recall (from my own projects) that there may be some other element types as well (e.g. picker wheels or file sharing modals). I will check this on Monday/Tuesday, but those can also be handled in a separate PR if necessary.

I don't remember any cases where springboard shows something different from an alert. Please provide examples of the system app page source for such cases.

**Comment by KazuCocoa** [comment]
I'm also curious about other cases.

So far, I have checked like below share features before. They were not managed by springboard, so the same app scope worked. Default photos, a couple of 3rd party apps. Actually, they could depend on the app implementation so this might not be the all cases.

Screenshot 2024-06-01 at 10 54 29 AM

Screenshot 2024-06-01 at 10 56 00 AM

After seeing the comment, I checked some data picker (`XCUIElementTypeDatePicker`, `XCUIElementTypePickerWheel`) implementations in a few apps like apple's calender and 3rd party apps I knew of, they worked in their own scope (so not springboard)

Screenshot 2024-06-01 at 11 43 26 AM

**Comment by eglitise** [comment]
I will check my project on Monday/Tuesday, but I don't think it should block this PR.

**Comment by eglitise** [comment]
Okay, I have checked a bit and found 2 cases where springboard places something other than an alert over the app:
* Floating Picture-in-Picture window
  * This case is interesting because the PiP is added to both the app and springboard XML sources, but only the springboard XML contains all the correct window details
  * XCUITest driver <6 did not detect this and remained in the same app
* TestFlight's 'From the Developer' and 'Share Feedback' screens - [here is an example with screenshots](https://support.readyeducation.com/hc/en-us/articles/1500011103342-Access-the-iOS-Beta-App)
  * XCUITest driver <6 detected this and switched to the springboard as the active app

I will have more time to investigate the XMLs tomorrow.

**Comment by eglitise** [comment]
@mykola-mokhnach here is a gist with XMLs for the two TestFlight screens mentioned above (edited for privacy): https://gist.github.com/eglitise/4704401148fef1db400e8e6be782973b

**Pull request merged**